src/NuGet.Clients/NuGet.CommandLine/Commands/InstallCommand.cs

```diff
@@ -186,7 +186,7 @@ private Task PerformV2Restore(string packagesConfigFilePath, string installPath)
                 Console.LogMinimal(message);
             }
 
-            Task<PackageRestoreResult> packageRestoreTask = PackageRestoreManager.RestoreMissingPackagesAsync(packageRestoreContext, new ConsoleProjectContext(Console));
+            Task<PackageRestoreResult> packageRestoreTask = PackageRestoreManager.RestoreMissingPackagesAsync(packageRestoreContext, new ConsoleProjectContext(Console), new SourceCacheContext());
```

Comment:
Yes this causes an issue for `nuget.exe install packages.config -DirectDownload ...`.